HW0/P5/P5.txt

```diff
@@ -1 +1,12 @@
 # Your answers here.
+1, 255 seconds
+2, 34 seconds. 8 people and each people takes 32 bags. It takes 31 seconds to finish the counting at the first stage. Furthremore, it takes log2(8) = 3 seconds to finish. So in total it is 31+3 = 34 seconds.
+3, 8 seconds
+4, done. Its log2(N)(round up to the nearest integer) seconds
+5, computing time = 8 seconds. However the communication time is 7 seconds, because at the first stage, the cashier would not need to communicate with himself. so in total it is 15 seconds.
+6, If each cashier comes and takes 1 bag. It takes 256 + 8 + 8 = 272 seconds.
```

Comment:
I got a different answer which quite likely is wrong, but I don't see where 256+8+8 come in. The 256 I assume is for handing out all the bags, where do the 8 and 8 come from?